### ISSUE TITLE
Suggestion: put it in apple's appstore for OS X to get more $$

### ISSUE THREAD

**Issue by GreatGranPa** [aardappel/treesheets#76]

**Comment by GreatGranPa** [comment]
It's now stable enough under OSX to charge a few bucks for it!

**Comment by aardappel** [comment]
Not so sure.. TreeSheets enjoys a medium level of popularity because it is free, but so far has not shown any commercial potential (very low amount of donations, for example). OS X also has the lowest amount of users of the 3 platforms. Apple is pretty restrictive on what is allowed on its app store, and I am not sure TreeSheets complies.

**Comment by DivineDominion** [comment]
Since this is a tool for productivity nerds, I think the Mac App Store isn't the best spot anyway. I always recommend selling outside the MAS. (Am a Mac dev, btw.)

However, charging happy Mac users money is still a good idea -- most of them happily pay for good software. Could help you raise funds in the long run. The UI may be a show stopper, though, attracting only super early adopters and believers in the app. My wild assed guess: this isn't worth the trouble right now, but you should keep this in mind as the Mac version matures.

**Comment by aardappel** [comment]
@DivineDominion thanks for the insight!

The UI is shared with the other two platforms, which means I don't have a lot of control over how it looks. A native mac UI would be a ton of extra work and maintenance, so as you say, probably not worth it.

I've had hopes that over time the mac port of wxWidgets would improve, but so far it hasn't much.

**Comment by aardappel** [comment]
Also not easy to charge for an app that is open source.

**Comment by DivineDominion** [comment]
I finished reading Steve Rosenberg's _Draming in Code_ a while ago. They used wxWidgets in the [case study](https://en.wikipedia.org/wiki/Chandler_(software)), too :) I am amazed that it's still stable, given all the changes Apple drives down programmers's throats over the years.

Regarding charging for open source: don't worry.

From a developer's standpoint, sure, you just grab the code and compile it yourself -- given that you know your way around native app programming, not just web frontend development. Making a paid app open source can be risky for various reasons, but making an open source app paid doesn't make much difference for those who rely on a binary download anyway. Eric Steven Raymond (author of _Cathedral and the Bazaar_) wrote about economics and OSS in [_The Magic Cauldron_](http://www.catb.org/~esr/writings/cathedral-bazaar/magic-cauldron/).

Or you make it a patronage model; paying for OSS apps is like patronage anyway.

Again, Mac users will not like the UI very much, so that's a thing, but in general, being open source doesn't mean people don't want to support the developer by throwing money at him/her 👍

**Comment by aardappel** [comment]
I agree, 99% of people would not avoid paying by making their own build. Though for some paid open source programs, alternative download sites tend to pop up.

If by patronage you mean donationware, it's already that: http://strlen.com/treesheets/docs/donations.html
I guess I could do a better job encouraging users to donate :)

But yeah, it be good to have a look if any of the biggest UI offenses can be worked around. If you're a mac user, what is worst about the UI?

**Comment by DivineDominion** [comment]
It's mostly subtle details, really, that affect the app's chrome. (The custom rendered nested table canvas doesn't look odd on its own; it's custom!)

Looking at the sample:

![The App](https://cloud.githubusercontent.com/assets/59080/25556647/579d9b40-2d00-11e7-99ae-642464805358.png)

This is what makes it look like an emulated window:

* boxes and text fields in the toolbar; but a single search field would be expected (in the top right corner then)
![Search](https://cloud.githubusercontent.com/assets/59080/25556691/2e85c09c-2d01-11e7-91f5-2fba3f085552.png)
* colored toolbar icons (nowadays they are mostly b/w with some color accent fills, see Safari and Numbers screenshots below)
* status bar has color (comparatively distractful), and could take a few pixels of extra whitespace at the top 
* gradient of the tab bar (the brighter color should be a bit darker to tone it down -- or be replaced with a flat darker color)
* tabs at the bottom are unusual
* Undo/Redo menu items should be at the top

Regarding the content/canvas itself:

* vertical whitespace is very limited, especially pronounced by the rather thick borders, so it feels cramped
* use system font or Helvetica instead of Arial (or is it Verdana?) so it doesn't look like a Windows app

If I can figure out how your code works, I'd be happy to help from time to time in between my own app coding frenzies. I can make this into a task list in a separate issue to organize contributions if you like.

---

Safari:  
![Safari](https://cloud.githubusercontent.com/assets/59080/25556665/90d43b8a-2d00-11e7-8e8d-c5d8850cd88f.png)

Numbers:  
![Numbers](https://cloud.githubusercontent.com/assets/59080/25556670/a5666988-2d00-11e7-940b-c3e63ef8adad.png)

MindNode:  
![MindNode](https://cloud.githubusercontent.com/assets/59080/25556711/a33d94d2-2d01-11e7-9234-436dc4eb6b16.png)

**Comment by aardappel** [comment]
Thanks! A couple of them would probably be easy fixes:

- Use a custom default font for mac.
- Find a set of mac-like b&w icons.
- Some special purpose mac-like spacing where possible.
- I can experiment with the tabs being at the top.. I remember that looking rather odd though.

Some other may be harder, as that requires piercing through the wxWidgets abstractions :)

**Comment by aardappel** [comment]
If you want to contribute, I'd play around with some of this low hanging fruit and see how far you get. The code should not be difficult to find your way around, as it is all pretty small and simple. There's an Xcode project you can use. You can make mac specific changes by surrounding them with `#ifdef __WXMAC__`
e.g.:

- toolbar creation around line 582 in myframe.h. Icon sets are being determined there too.
- default font line 90 in system.h
- tabs at bottom 659 in myframe.h

**Comment by gedw99** [comment]
Just curious but is Apple that picky about the GUI look and feel for apps sold through the app store ?

I did a mini search.
Slack which looks nothing like MacOS is in their store.
https://itunes.apple.com/de/app/slack/id803453959?l=en&mt=12

Pat of the reason i ask is for me. I am using Material Design style for Desktop apps, and would hate it to be rejected by Apple because of that.

**Comment by DivineDominion** [comment]
Nope, that's usually not a problem. It's just that users are often (too) picky :)
